### ISSUE TITLE
Order for CSS Pesudoselector not

### ISSUE THREAD

**Issue by jeremylt** [freeCodeCamp/freeCodeCamp#46003]
### Describe the Issue

It is introduced here:

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-typography-by-building-a-nutrition-label/step-48

But it is used prior to the introduction here:

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-forms-by-building-a-registration-form/step-41

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-forms-by-building-a-registration-form/step-41

### Your code

N/A

### Expected behavior

Material should be introduced before used.

### System

All

**Comment by hvinnieg** [comment]
I'll see if I can fix this I'll let u know how it goes

**Comment by jeremylt** [comment]
I'm not sure what fix we want yet - the curriculum folks haven't chimed in yet.

**Comment by hvinnieg** [comment]
Yea I was waiting to

**Comment by Sboonny** [comment]
I think we can move instructions of the `selector` to **step 41**, while altering the description text to drive focus into using said `selector`, because the challenges require `not` and if you use different `selector` like the one in the bottom. it will fail.

```css
fieldset:nth-last-child(1n+2){
  border-bottom: 3px solid #3b3b4f
}
```

**Comment by jeremylt** [comment]
I like this approach

**Comment by moT01** [comment]
Let's see if we can get one more person to approve and then we can either refine the wording, or open it up for help and refine the wording on the PR.

**Comment by Sboonny** [comment]
> First, I would like to mention that these are two different projects - I didn't notice right away.
> 
> One issue I see is that step 41 is trying to introduce two new concepts at once. `:not` and `:last-of-type`. I don't see either of them being used before this - or any pseudo selectors/classes for that matter. It's probably confusing to jump to those two selectors right away with no previous mentions of any other pseudo-selectors.
> 
> Here's how I would fix it - change step 41 to add a border-bottom to all of the `fieldset` elements:
> 
> ```
> To give the `fieldset` elements a bit of separation,
> add a `border-bottom` of `3px solid #3b3b4f` to all of them.
> ```
> 
> And they have to add that border-bottom to the existing fieldset selector.
> 
> And add a new step after 41:
> 
> ```
> The last border looks a little out of place. You can select the last
> element of a specific type using the `last-of-type` CSS pseudo-class, like this: 
> 
> ```
> p:last-of-type { }
> ```
> 
> That will select the last `p` element. Create a new selector that targets the
> last `fieldset` element and set its `border` to `none`.
> ```
> 
> And we could maybe add a "That looks better." at the start of the current step 42.
> 
> That's what I'm thinking anyway - and we can leave step 48 alone then.

Changing steps 41 and 42 like this, sound reasonable to me. I add my voice to Tom suggestion 🤝

**Comment by lasjorg** [comment]
> That will select the last `p` element.

We might have to qualify that a bit. Here is how MDN words it.

> represents the last element of its type among a group of sibling elements.

---

So are we talking about something like this?

41. Have them add the border to all fieldset elements.

42. Have them remove the border from the last fieldset element using `:last-of-type`

43. Have them combine 41 and 42 using `:not(:last-of-type)`

**Comment by moT01** [comment]
No, your step 43 wouldn't be there @lasjorg. The other two are correct.

41: add border to all `fieldeset` elements with `border: 3px solid #3b3b4f`
42: remove border from last element with:
```
fieldset: last-of-type { 
  border: none;
}
```
43: would be the current step 42.

And the `:not` just wouldn't be there anymore.

**Comment by lasjorg** [comment]
That is how I understood it initially as well. I'm guessing adding a new challenge wouldn't be super easy to do. Although, removing the `:not()` selector also affects all the seed code.

I just think that it would be a shame not to show the `:not(:last-of-type)` as it's such a common task applying the same styles to all but the first or last sibling elements. And it is made a lot cleaner using `:not(:last-of-type)` compared to applying the styles to all sibling elements and then removing the styles from the first or last.

I did a search for the selector and it doesn't seem to be used anywhere else in the curriculum. I did find the `:not()` selector used in the balance sheet challenge so that's good. But it still is a bummer to have to completely remove the `:not(:last-of-type)` just because it wasn't introduced correctly.

---

What does creating a new step involve? Do you just create a new step and then bump the steps in the `title` and `dashedName` for the following steps or is there more to it?

**Comment by moT01** [comment]
Adding a new challenge with my changes would require changing the seed code on all steps after it. So no, it wouldn't be super easy - but we do have some [helper tools for adding new steps](https://contribute.freecodecamp.org/#/how-to-work-on-practice-projects), among other things. I believe they will add the step, and bump the title and whatever else on all the other ones after it. So that would reduce the amount of work quite a bit.

> I just think that it would be a shame not to show the `:not(:last-of-type)`

I'm not opposed to leaving that selector, but I think we need to find another way to reword things so they're easier to grasp. I suppose we could just change my suggested 42 to adding the `:not(:last-of-type)`. I don't think that's too much. Particularly since you say that are commonly used together. I would be fine with that. 

41: add border to all `fieldset`
42: add `:not(:last-of-type)`

**Comment by lasjorg** [comment]
> Adding a new challenge with my changes would require changing the seed code on all steps after it.

So does removing the `:not()` selector we have to remove it from all the following seed code. I'm not sure I understand what happens to the current step 42, wouldn't that need to get pushed forward as well?

---

Explaining both selectors in one step without making the step too wordy will be challenging.

I know this is too long but I can't really figure out how to condense it.

Step 41

> To give the `fieldset` elements a bit of separation we want to give all, except the last one, a `border-bottom`.
> 
> One option is to set the style on all the `fieldset` elements and then remove it from the last one using `:last-of-type` . The `:last-of-type` _pseudo-class_ selector selects the last element of its type among a group of sibling elements. For example, `fieldset:last-of-type` would select the last `fieldset` element.
> 
> But what if instead, we just selected all the elements, except the last. For this, we can use the `:not()` _pseudo-class_ selector and combine it with `:last-of-type`. For example, `p:not(:last-of-type)` would select all the `p` elements, except the last.
> 
> Select all but the last `fieldset` element, and give them a `border-bottom` of `3px solid #3b3b4f`.
